src/main/kotlin/com/hypto/iam/server/service/SsoLoginSevice.kt

```diff
@@ -0,0 +1,33 @@
+package com.hypto.iam.server.service
+
+import com.hypto.iam.server.configs.AppConfig
+import com.hypto.iam.server.models.AuthUrlResponse
+import com.hypto.iam.server.models.SsoLoginRequest
+import com.workos.WorkOS
+import com.workos.organizations.OrganizationsApi
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class SsoLoginServiceImpl : SsoLoginService, KoinComponent {
+    private val appConfig: AppConfig by inject()
+
+    override suspend fun getAuthUrlForDomain(
+        ssoLoginRequest: SsoLoginRequest,
+    ): AuthUrlResponse {
+        val workos = WorkOS(appConfig.workOS.secretKey)
+        val domains = listOf(ssoLoginRequest.domain)
+        val options = OrganizationsApi.ListOrganizationsOptions.builder().domains(domains).build()
+        val organizations = workos.organizations.listOrganizations(options)
```

Comment:
error handling and error message transformation for any exceptions that the workos client throws seems to be missing

Reply:
errors handled now